source/crud/tests/unified/replaceOne-dots_and_dollars.yml

```diff
@@ -0,0 +1,180 @@
+    runOnRequirements:
+      - minServerVersion: "5.0"
+    operations:
+      - name: replaceOne
+        object: *collection0
+        arguments:
+          filter: { _id: 1 }
+          replacement: &dollarPrefixedKeyInEmbedded { _id: 1, a: { $b: 1 } }
+        expectResult: *replaceResult
+    expectEvents: &expectEventsDollarPrefixedKeyInEmbedded
+      - client: *client0
+        events:
+          - commandStartedEvent:
+              command:
+                update: *collection0Name
+                updates:
+                  - q: { _id: 1 }
+                    u: *dollarPrefixedKeyInEmbedded
+                    multi: { $$unsetOrMatches: false }
+                    upsert: { $$unsetOrMatches: false }
+    outcome:
+      - collectionName: *collection0Name
+        databaseName: *database0Name
+        documents:
+          - *dollarPrefixedKeyInEmbedded
+
+  - description: "Replacing document with dollar-prefixed key in embedded doc on pre-5.0 server yields server-side error"
+    runOnRequirements:
+      - maxServerVersion: "4.99"
+    operations:
+      - name: replaceOne
+        object: *collection0
+        arguments:
+          filter: { _id: 1 }
+          replacement: *dollarPrefixedKeyInEmbedded
+        expectError:
+          isClientError: false
+    expectEvents: *expectEventsDollarPrefixedKeyInEmbedded
+    outcome: *initialData
+
+  - description: "Replacing document with dotted key in embedded doc on 3.6+ server"
+    runOnRequirements:
+      - minServerVersion: "3.6"
+    operations:
+      - name: replaceOne
+        object: *collection0
+        arguments:
+          filter: { _id: 1 }
+          replacement: &dottedKeyInEmbedded { _id: 1, a: { b.c: 1 } }
+        expectResult: *replaceResult
+    expectEvents: &expectEventsDottedKeyInEmbedded
+      - client: *client0
+        events:
+          - commandStartedEvent:
+              command:
+                update: *collection0Name
+                updates:
+                  - q: { _id: 1 }
+                    u: *dottedKeyInEmbedded
+                    multi: { $$unsetOrMatches: false }
+                    upsert: { $$unsetOrMatches: false }
+    outcome:
+      - collectionName: *collection0Name
+        databaseName: *database0Name
+        documents:
+          - *dottedKeyInEmbedded
+
+  - description: "Replacing document with dotted key in embedded doc on pre-3.6 server yields server-side error"
+    runOnRequirements:
+      - maxServerVersion: "3.4.99"
+    operations:
+      - name: replaceOne
+        object: *collection0
+        arguments:
+          filter: { _id: 1 }
+          replacement: *dottedKeyInEmbedded
+        expectError:
+          isClientError: false
+    expectEvents: *expectEventsDottedKeyInEmbedded
+    outcome: *initialData
+
+  - description: "Unacknowledged write using dollar-prefixed or dotted keys may be silently rejected on pre-5.0 server"
```

Comment:
This corresponds to the similarly-named test in `insertOne-dots_and_dollars.yml`.